raiden/api/rest.py

```diff
@@ -891,8 +891,34 @@ def leave(self, registry_address: typing.PaymentNetworkID, token_address: typing
         return api_response(result=closed_channels)
 
     @requires_api_key
-    def register_secret_light(self, signed_tx: typing.SignedTransaction):
+    def register_secret_light(self, internal_msg_identifier: int, signed_tx: typing.SignedTransaction):
         try:
+            message = LightClientMessageHandler.get_light_client_protocol_message_by_internal_identifier(
+                internal_msg_identifier=internal_msg_identifier,
+                wal=self.raiden_api.raiden.wal
+            )
+
+            if not message:
+                return ApiErrorBuilder.build_and_log_error(
+                    errors="Light Client Message with internal_msg_identifier = {} not found"
+                        .format(internal_msg_identifier),
+                    status_code=HTTPStatus.BAD_REQUEST,
+                    log=log
+                )
+
+            if message.is_signed:
+                return ApiErrorBuilder.build_and_log_error(
+                    errors="Message already signed",
+                    status_code=HTTPStatus.CONFLICT,
+                    log=log
+                )
```

Comment:
thanks for changing this :clap: 

would you mind also having it this way in the `settlement_light` function? 

i'm asking because it's in the same file and i don't think it's too big of a change not to have it. 

it would help consistency, i think